src/windows/process.rs

```diff
@@ -559,35 +559,37 @@ pub(crate) fn get_system_computation_time() -> ULARGE_INTEGER {
 }
 
 pub(crate) fn get_disk_usage(p: &mut Process){
-    let r = ProcessDiagnosticInfo::try_get_for_process_id(p.pid as u32).ok();
-    if r.is_some(){
-        let r = r.unwrap();
-        if r.is_some(){
-            let du = r.unwrap().get_disk_usage().ok();
-            if du.is_some(){
-                let du = du.unwrap();
-                if du.is_some(){
-                    let report = du.unwrap().get_report().ok();
-                    if report.is_some(){
-                        let report = report.unwrap();
-                        if report.is_some(){
-                            let report = report.unwrap();
-                            let read_bytes = report.get_bytes_read_count().ok();
-                            let write_bytes = report.get_bytes_written_count().ok();
-                            if read_bytes.is_some(){
-                                p.read_bytes = read_bytes.unwrap() as u64;
-                            }
-                            if write_bytes.is_some(){
-                                p.written_bytes = write_bytes.unwrap() as u64;
-                            }
-                        }
-                    }
-                }
-            }
-            
-        }
-        
-    }
+    let diag_info = ProcessDiagnosticInfo::try_get_for_process_id(p.pid as u32).ok();
+    match diag_info{
+        Some(diag_info) => match diag_info{
```

Comment:
Isn't of this match cascade, you can add a new macro:

```rust
macro_rules! safe_unwrap {
    ($x:expr) => {
        match $x {
             Some(x) => x,
             None => return,
        }
    }
```

And you can use it like this:

```rust
let diag_info = safe_unwrap!(diag_info.get_disk_usage().ok());
```